src/core/utils/components/MiddleTextTruncation.tsx

```diff
@@ -0,0 +1,61 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import { useOverflow } from '../hooks/useOverflow';
+import { CommonProps } from '../props';
+
+const ELLIPSIS_CHAR = '…';
+
+export type MiddleTextTruncationProps = {
+  /**
+   * Text to truncate.
+   */
+  text: string;
+  /**
+   * Number of characters to leave at the end.
+   * @default 6
+   */
+  endCharsCount?: number;
+} & CommonProps;
+
+/**
+ * Truncates text with the ellipsis in the middle,
+ * leaving defined number of chars at the end.
+ * @example
+ * <MiddleTextTruncation text='Here can be a very long text' />
```

Comment:
Maybe show a more useful example here, like a file name with numbers/extension at the end

Reply:
Updated :D